### PR TITLE
Remove developer who removed github account

### PR DESCRIPTION
**Pull request by howardjohn**
https://github.com/domgoer is now a 404

The most common reason to send a PR to this repository is to join the Istio org.  If you are joining, please fill out this template.  If you are not, please **delete all the below**.

- [ ] I have reviewed the [community membership guidelines](https://github.com/istio/community/blob/master/ROLES.md#member)
- [ ] I have reviewed the [contribution guidelines](https://github.com/istio/community/blob/master/CONTRIBUTING.md)
- [ ] I have enabled [2FA on my GitHub account](https://github.com/settings/security)
- [ ] I have joined the [Istio discussion board](https://discuss.istio.io) and subscribed to the [Contributors](https://discuss.istio.io/c/contributors) category
- [ ] I have joined [Istio's Slack workspace](https://slack.istio.io)

List your company name, or indicate Individual if you're not affiliated with a company:

Provide a link to at least one PR that you have successfully pushed to one of the Istio repos:

### CODE REVIEW

**Comment by craigbox** [comment]
Appears to have been renamed to domechn

File: org/developers.yaml [hunk 1]
```diff
@@ -41,7 +41,6 @@ developers:
 - dgn
 - diemtvu
 - dmitri-d
```
Comment (craigbox) [inline org/developers.yaml:43]:
```suggestion
- dmitri-d
- domechn
```

**Review by craigbox — CHANGES_REQUESTED** [review]

File: org/developers.yaml (hunk 1 above)
Comment (craigbox) [reply org/developers.yaml:43]: @domechn to confirm

**Review by craigbox — APPROVED** [review]

**Comment by nrjpoddar** [comment]
@howardjohn how are you finding inactive/deleted accounts? Is there automation that surfaces these accounts?

**Comment by howardjohn** [comment]
The postsubmit job fails and sends a notification to slack/email

File: org/developers.yaml (hunk 1 above)
Comment (domechn) [reply org/developers.yaml:43]: confirmed, thanks for your modification

**Pull request merged**